Add hyperlink support for URLs in XCom values

dheerajturaga opened the pull request:
XCom values containing URLs are now automatically detected and rendered as clickable hyperlinks that open in new tabs with proper security attributes.

<img width="1602" height="456" alt="image" src="https://github.com/user-attachments/assets/140dad7f-17c0-4bb0-a19a-7ea39f6483c6" />

dheerajturaga committed:
Add hyperlink support for URLs in XCom values

XCom values containing URLs are now automatically detected and rendered
as clickable hyperlinks that open in new tabs with proper security attributes.

pierrejeambrun reviewed (commented):
There is already a `urlRegex` in `constants/urlRegex.ts` that we should reuse.

You can take example at `pages/Providers` because this is really similar to what we are trying to achieve here.

dheerajturaga committed:
Refactor XCom URL hyperlink to use existing urlRegex constant

Replaced custom URL regex and validation logic with the shared urlRegex
from constants/urlRegex.ts, following the same pattern used in the
Providers page for consistency across the codebase.

pierrejeambrun reviewed (approved):
Cool thanks.
